### PR TITLE
typings: support Raw types for insert, where, update

### PR DESCRIPTION
**Pull request by maximelkin**
Closes https://github.com/knex/knex/issues/3652

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -678,8 +688,8 @@ declare namespace Knex {
       >[]
     >(
       columnName: K1,
-      value: TRecord[K1],
-      returning: K1
+      value: DbColumn<TRecord[K1]>,
+      returning: K2
```
Comment (maximelkin) [inline types/index.d.ts]: side fix

File: types/index.d.ts [hunk 2]
```diff
@@ -691,12 +701,12 @@ declare namespace Knex {
       >[]
     >(
       columnName: K1,
-      value: TRecord[K1],
-      returning: readonly K1[]
+      value: DbColumn<TRecord[K1]>,
+      returning: readonly K2[]
```
Comment (maximelkin) [inline types/index.d.ts]: side fix

**Comment by maximelkin** [comment]
@kibertoad this can be merged, since this is not breaking change

**Comment by lorefnon** [comment]
@maximelkin Thanks for this PR. The changes look good :) 

The only minor comment I have is on the naming: `DbRecordSet` -> `DbRecordArr` to convey that it is not actually a [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set).

**Comment by maximelkin** [comment]
Set in name means not data structure, but some amount of items
And this is not always array

**Comment by maximelkin** [comment]
also, what do you think about renaming DbColumn to DbAttribute?

**Comment by kibertoad** [comment]
@lorefnon Is this good enough to be merged already?

**Comment by lorefnon** [comment]
Yeah, I think the current set of changes look good. 

I don't have a preference around DbColumn vs DbAttribute.

**Pull request merged**